sql/core/src/main/scala/org/apache/spark/sql/execution/Generate.scala

```diff
@@ -51,9 +52,12 @@ case class Generate(
     join: Boolean,
     outer: Boolean,
     output: Seq[Attribute],
+    generatorOutput: Seq[Attribute],
     child: SparkPlan)
   extends UnaryNode {
 
+  override def missingInput: AttributeSet = super.missingInput -- generatorOutput
+
```

Comment:
You can use the same approach in logical.Generate, i.e.,

     override def expressions: Seq[Expression] = generator :: Nil

to solve this issue. Then you don't need to modify SparkStrategies.